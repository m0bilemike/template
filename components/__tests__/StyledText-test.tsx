import { render } from '@testing-library/react-native';
import React from 'react';
import { MonoText } from '../StyledText';

// Mock the Themed module so we don't depend on theme logic in this test
jest.mock('../Themed', () => {
  const React = require('react');
  const { Text } = require('react-native');
  return {
    Text: ({ children, style, ...rest }: any) => (
      <Text testID="mono-text" style={style} {...rest}>
        {children}
      </Text>
    ),
  };
});

describe('<MonoText />', () => {
  it('renders its children', () => {
    const { getByText } = render(<MonoText>Snapshot test!</MonoText>);
    expect(getByText('Snapshot test!')).toBeTruthy();
  });

  it('applies the SpaceMono font family', () => {
    const { getByTestId } = render(<MonoText>Test</MonoText>);
    const element = getByTestId('mono-text');
    const style = Array.isArray(element.props.style)
      ? Object.assign({}, ...element.props.style)
      : element.props.style;

    expect(style.fontFamily).toBe('SpaceMono');
  });

  it('renders consistently (snapshot)', () => {
    const { toJSON } = render(<MonoText>Snapshot test!</MonoText>);
    expect(toJSON()).toMatchSnapshot();
  });
});