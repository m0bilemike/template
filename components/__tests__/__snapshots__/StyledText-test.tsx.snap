// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonoText /> renders consistently (snapshot) 1`] = `
<Text
  style={
    [
      undefined,
      {
        "fontFamily": "SpaceMono",
      },
    ]
  }
  testID="mono-text"
>
  Snapshot test!
</Text>
`;

exports[`renders correctly 1`] = `null`;
